pkg/etcd/etcd.go

```diff
@@ -0,0 +1,17 @@
+package etcd
+
+import "strings"
+
+// Prefix for all etcd records.
+var GlobalEtcdPrefixPath string
+
+func EtcdPath(key string) string {
+	if GlobalEtcdPrefixPath == "" || strings.HasPrefix(key, GlobalEtcdPrefixPath) {
+		return key
+	}
+	if strings.HasPrefix(key, "/") {
+		return GlobalEtcdPrefixPath + key
+	} else {
+		return GlobalEtcdPrefixPath + "/" + key
+	}
+}
```

Comment:
This is a bad implementation @ihostage I will send a proper implementation in a bit

Reply:
Yeeeees... I didn’t like it either. I do not know Golang and structure of MinIO very well. It was a quick resolving of the cycle of imports. I tried to create something working for initiate discussion.